platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,210 @@
+    // ServletContextHandler
+    Optional<ServletContextHandler> optionalHttpContext = getHTTPContext(service, refBundle);
+
+    if (!optionalHttpContext.isPresent()) return;
+
+    ServletContextHandler httpContext = optionalHttpContext.get();
+
+    ErrorPageErrorHandler errorPageErrorHandler = null;
+    try {
+      errorPageErrorHandler = (ErrorPageErrorHandler) httpContext.getErrorHandler();
+
+    } catch (ClassCastException e) {
+      LOGGER.warn(
+          "Problem getting the ErrorHandler from the ServletContextHandler for the bundle {}.",
+          refBundle.getSymbolicName());
+    }
+
+    if (errorPageErrorHandler != null && errorPageErrorHandler.getErrorPages().isEmpty()) {
+
+      createAndAddErrorPageHandler(httpContext, refBundle);
+    }
+  }
+
+  private void injectErrorPage(ServletContext context, Bundle refBundle) {
+
+    Optional<ServletContextHandler> optionalHttpServiceContext = getHTTPContext(context, refBundle);
+    optionalHttpServiceContext.ifPresent(p -> createAndAddErrorPageHandler(p, refBundle));
+  }
+
+  private Optional<ServletContextHandler> getHTTPContext(ServletContext context, Bundle refBundle) {
+
+    Field field;
+    try {
+      // this grabs the enclosing instance class, which is actually a private class
+      // this is the only way to do this in Java
+      field = context.getClass().getDeclaredField("this$0");
+      field.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      LOGGER.warn(
+          "Unable to find enclosing class of ServletContext for delegating the error page. The default jetty errors will display in the browser",
+          e);
+
+      return Optional.empty();
+    }
+
+    try {
+      // need to grab the servlet context handler so we can get down to the handler, which is what
+      // we really need
+      return Optional.of((ServletContextHandler) field.get(context));
+
+    } catch (IllegalAccessException e) {
+      LOGGER.warn(
+          "Unable to get the ServletContextHandler for {}. Jetty's default error page will be used for this context",
+          refBundle.getSymbolicName(),
+          e);
+
+      return Optional.empty();
+    }
+  }
+
+  private void createAndAddErrorPageHandler(
+      ServletContextHandler httpServiceContext, Bundle refBundle) {
+
+    // now that we have the handler, we can add in our own ErrorServlet
+    ServletHandler handler = httpServiceContext.getServletHandler();
+
+    ServletHolder errorServletHolder = new ServletHolder(new ErrorServlet());
+    errorServletHolder.setServletHandler(handler);
+
+    try {
+      errorServletHolder.start();
+      errorServletHolder.initialize();
+    } catch (Exception e) {
+      LOGGER.warn(
+          "Unable to initialize an errorServletHolder for {}. Jetty's default error page will be used for this context",
+          refBundle.getSymbolicName(),
+          e);
+
+      return;
+    }
+
+    LOGGER.info(
+        "Injecting an error page into into {} ID: {}",
```

Comment:
✏️  have `into` 2x in this log statement

Reply:
👍